Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/CallBAP.hs

```diff
@@ -154,12 +160,30 @@ sendRideAssignedUpdateToBAP booking ride = do
       >>= fromMaybeM (MerchantNotFound booking.providerId.getId)
   driver <- QPerson.findById ride.driverId >>= fromMaybeM (PersonNotFound ride.driverId.getId)
   vehicle <- QVeh.findById ride.driverId >>= fromMaybeM (VehicleNotFound ride.driverId.getId)
+  imagePath <- ((.compAadhaarImagePath) =<<) <$> CQDI.findById (cast ride.driverId)
+  image <- case imagePath of
+    Nothing -> return Nothing
+    Just path ->
+      fetchDriverAadhaarImage ride.driverId >>= \case
+        Just img -> return $ Just img
+        Nothing -> Just <$> (cacheDriverImageByDriverId ride.driverId /=<< S3.get (unpack path))
   let rideAssignedBuildReq = ACL.RideAssignedBuildReq {..}
   rideAssignedMsg <- ACL.buildOnUpdateMessage rideAssignedBuildReq
 
   retryConfig <- asks (.shortDurationRetryCfg)
 
   void $ callOnUpdate transporter booking.bapId booking.bapUri booking.bapCity booking.bapCountry booking.transactionId rideAssignedMsg retryConfig
+  where
+    driverAadhaarImageKey :: Id SP.Person -> Text
+    driverAadhaarImageKey driverId = "Driver-Compressed-Aadhaar-Image-Key:DriverId" <> getId driverId
+
+    fetchDriverAadhaarImage :: (CacheFlow m r) => Id SP.Person -> m (Maybe Text)
+    fetchDriverAadhaarImage driverId = Hedis.withCrossAppRedis (Hedis.safeGet $ driverAadhaarImageKey driverId)
```

Comment:
Why are you using **Hedis.withCrossAppRedis** ?

Reply:
Removed